central/networkgraph/service/service_impl.go

```diff
@@ -101,24 +100,60 @@ func (s *serviceImpl) GetExternalNetworkEntities(ctx context.Context, request *v
 	}
 
 	query, _ = search.FilterQueryWithMap(query, schema.NetworkEntitiesSchema.OptionsMap)
-	pred, err := netEntityPredFactory.GeneratePredicate(query)
+
+	ets, err := s.entities.GetEntityByQuery(ctx, query)
```

Comment:
I don't like this abbreviation. Perhaps change `ets` to `entities`. In that case you should change `s.entities` to `s.entitiesDS`.

Reply:
You could also change it to ents, they might help you fight Sauron...

![image](https://github.com/user-attachments/assets/66589f34-0942-4e14-b9e5-1850f3f4b172)